parsl/executors/workqueue/executor.py

```diff
@@ -184,51 +192,74 @@ def WorkQueueSubmitThread(task_queue=multiprocessing.Queue(),
                 collector_queue.put_nowait(msg)
                 continue
 
-            logger.debug("Task {} submitted workqueue with id {}".format(parsl_id, wq_id))
+            logger.debug("Task {} submitted WorkQueue with id {}".format(parsl_id, wq_id))
 
         if cancel_value.value == 0:
             continue_running = False
 
-        # Wait for Tasks
+        # If the queue is not empty wait on the WorkQueue queue for a task
         task_found = True
-        # If the queue is not empty wait on the workqueue queue for a task
         if not q.empty() and continue_running:
             while task_found is True:
                 if cancel_value.value == 0:
                     continue_running = False
                     task_found = False
                     continue
+
+                # Obtain the task from the queue
                 t = q.wait(1)
                 if t is None:
                     task_found = False
                     continue
                 else:
                     parsl_tid = t.tag
-                    logger.debug("Completed workqueue task {}, parsl task {}".format(t.id, parsl_tid))
+                    logger.debug("Completed WorkQueue task {}, parsl task {}".format(t.id, parsl_tid))
                     status = t.return_status
                     task_result = t.result
                     msg = None
 
+                    # Task failure
                     if status != 0 or (task_result != WORK_QUEUE_RESULT_SUCCESS and task_result != WORK_QUEUE_RESULT_OUTPUT_MISSING):
-                        if task_result == WORK_QUEUE_RESULT_SUCCESS:
-                            logger.debug("Workqueue task {} failed with status {}".format(t.id, status))
-
+                        logger.debug("Wrapper Script status: {}\nWorkQueue Status: {}".format(status, task_result))
+                        # Wrapper script failure
+                        if status != 0:
+                            logger.debug("WorkQueue task {} failed with status {}".format(t.id, status))
                             reason = "Wrapper Script Failure: "
                             if status == 1:
-                                reason += "command line parsing"
-                            if status == 2:
+                                reason += "problem parsing command line options"
+                            elif status == 2:
                                 reason += "problem loading function data"
-                            if status == 3:
+                            elif status == 3:
                                 reason += "problem remapping file names"
-                            if status == 4:
+                            elif status == 4:
```

Comment:
it would be a bit defensive to stick a final "else" on the end here that says something like "i don't know what the reason was" that will deal with future changes adding on wrapper status numbers rather than leaving the reason blank.